src/components/organisms/Accordion/AccordionHeader/AccordionHeader.tsx

```diff
@@ -29,34 +31,34 @@ const Title = styled(Text)`
 `;
 
 export interface IAccordionHeader {
-  /** determines the position of the chevron icon */
-  isOpen: boolean;
+  id: string;
+  index: number;
   textSize: ITextProps['size'];
 }
 
-const AccordionHeader: FunctionComponent<IAccordionHeader> = React.forwardRef<HTMLButtonElement, IAccordionHeader>(
-  ({ children, isOpen, textSize = 'medium', ...rest }, ref) => {
-    const mapTextToArrowSize = {
-      large: '14px',
-      medium: '10px',
-      small: '8px',
-    };
-
-    return (
-      <StyledButton ref={ref} {...rest}>
-        <TitleContainer>
-          <Arrow
-            direction={isOpen ? 'down' : 'right'}
-            width={mapTextToArrowSize[textSize]}
-            height={mapTextToArrowSize[textSize]}
-          />
-          <Title weight="bold" size={textSize}>
-            {children}
-          </Title>
-        </TitleContainer>
-      </StyledButton>
-    );
-  },
-);
+const mapTextToArrowSize = {
+  large: '14px',
+  medium: '10px',
+  small: '8px',
+};
+
+const AccordionHeader: FC<IAccordionHeader> = ({ children, id, index, textSize = 'medium', ...rest }) => {
+  const { getHeaderProps, isActiveSection } = useAccordionContext();
+  const { ref, ...headerPropsRest } = getHeaderProps(id, index);
+  return (
+    <StyledButton ref={ref} {...headerPropsRest} {...rest}>
```

Comment:
I believe `<StyledButton />` doesn't need the `ref`, same for 
```js
<div ref={ref}>
``` 
in `<Accordion.Section />` ? 🤔

Reply:
it needs it to properly handle focus